### PR TITLE
Flipping the Board

### PR DESCRIPTION
**Pull request by mrwhalen22**
- Allows for the board to be seen from either the white or black perspective
- Refactors some board code from main into its own class #17 
image

### CODE REVIEW

**Review by Tacho2 — COMMENTED** [review]
Looks good, I appreciate the board class

**Review by Tacho2 — DISMISSED** [review]

File: src/board.cpp [hunk 1]
```diff
@@ -296,100 +185,99 @@ bool Board::isMoveValid(Move moveObject)
     std::tie(startPosX, startPosY) = moveObject.getStartTile();
     std::tie(endPosX, endPosY) = moveObject.getEndTile();
     ChessPieceType piece = moveObject.getPieceType();
+    bool isOppPiece = isOpposingPiece(moveObject);
+    bool isTeamPiece = isOwnPiece(moveObject);
+    bool isOccupied = isOppPiece || isTeamPiece;
+    int distanceX = std::abs(endPosX - startPosX);
+    int distanceY = std::abs(endPosY - startPosY);
```
Comment (mrwhalen22) [inline src/board.cpp:192]: Love the simplification. Should make the rest of the implementations easier

**Review by Tacho2 — APPROVED** [review]
I'M SO GLAD YOURE GOOD AT CPP (was implementing this earlier and it went to shit, somehow created a runtime error)

Great implementation, much better than how I was doing it

**Comment by Tacho2** [comment]
should we merge?

**Comment by mrwhalen22** [comment]
Im gonna merge this now, but Anna should be coming in soon with some changes for checking and checkmates

**Pull request merged**